buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy

```diff
@@ -131,6 +133,7 @@ class ClusterFormationTasks {
     /** Adds a task to extract the elasticsearch distribution */
     static Task configureExtractTask(String name, Project project, Task setup, NodeInfo node) {
         List extractDependsOn = [project.configurations.elasticsearchDistro, setup]
+        // TODO assert that project.configurations.elasticsearchDistro.singleFile is from a project and not the cache
```

Comment:
I did look at the direct project dependency thing but had a lot of trouble
which may have been caused by something I later fixed. So probably that is
possible. We could also only do the test if we were in an es checkout which
I think is a bit cleaner because then we just do less things when used
outside the checkout rather than different things?
On Nov 25, 2015 8:13 PM, "Ryan Ernst" <notifications@github.com> wrote:

> In
> buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy
> <https://github.com/elastic/elasticsearch/pull/15031#discussion_r45936254>
> :
>
> > @@ -131,6 +133,7 @@ class ClusterFormationTasks {
> >      /** Adds a task to extract the elasticsearch distribution */
> >      static Task configureExtractTask(String name, Project project, Task setup, NodeInfo node) {
> >          List extractDependsOn = [project.configurations.elasticsearchDistro, setup]
> > +        // TODO assert that project.configurations.elasticsearchDistro.singleFile is from a project and not the cache
>
> Perhaps we can just be smarter here. If we "knew" in the method that sets
> up distribution that we are in elasticsearch project, we can set up a
> direct project dependency instead of the current dependency.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/elastic/elasticsearch/pull/15031/files#r45936254>.
>